sdk/state/asset.go

```diff
@@ -0,0 +1,34 @@
+package state
+
+import (
+	"strings"
+
+	"github.com/stellar/go/txnbuild"
+)
+
+type Asset string
+
+const NativeAsset = Asset("native")
+
+func (a Asset) Native() bool {
```

Comment:
💭 to me `Native()` just based off the name feels like it's converting an asset to a Native representation. I remember you saying `Is<foo>` is generally not used in go, but I wonder in this case if it makes sense to just mimic the `IsNative()` method since that's what's currently on txnbuild.Asset as well?

I don't feel terribly strongly about this though if you think `Native()` is correct

Reply:
+1. I think IsX would be clearer here. When I said that IsX isn't common in Go, I was wrong about that. It is true that GetX/SetX naming isn't common and is explicitly called out in the effective Go docs, but IsX is common, even in the Go stdlib it is present. Updated in 5456314.

Reply:
> When I said that IsX isn't common in Go, I was wrong about that. It is true that GetX/SetX naming isn't common

oh I may have mixed those up 😅. good to know though thx 👍

Reply:
No you had heard me correctly, I was just wrong. I think at some point I started to believe no-Get/Set meant no-Is, but that's just not the case, which is good because the IsX reads better here.